### PR TITLE
Automatic splitting fine-tuning

### PR DESCRIPTION
**Pull request by matz-e**
Some changes put together:

* Change units of the automatic splitting to minutes, consistent with other time measures used in CRAB (as mentioned in #5623)
* Allow `Data.totalUnits` on the client to be fractional for all analysis splitting (needs client PR, more testing)
* Add bits to grab the status of the SubDAGs for a better status message on the client (also needs client PR)

Further, I've tried to improve logging when resubmitting on the schedd, which should allow for better analysis what is going on when resubmitting (snippet from `adjust_out.txt`):
```
…
2018-01-23 09:43:58.951511: Matchting the line 'DAG Node: Job1-203' to groups: ('1-203', '-203')
2018-01-23 09:43:58.951516: Successful match: 1-203, adjusting status and appending to adjustedJobIds
2018-01-23 09:43:58.952145: Matchting the line 'DAG Node: Job1-25' to groups: ('1-25', '-25')
2018-01-23 09:43:58.952213: Matchting the line 'DAG Node: Job1-103' to groups: ('1-103', '-103')
2018-01-23 09:43:58.952285: Matchting the line 'DAG Node: Job1-26' to groups: ('1-26', '-26')
2018-01-23 09:43:58.954310: Adjusting retries for job ids: ['1-20', '1-231', '1-248', '1-288', '1-215', '1-247', '1-287', '1-229', '1-285', '1-270', '1-261', '1-227', '1-371', '1-384', '1-383', '1-34', '1-132', '1-207', '1-152', '1-28', '1-180', '1-203']
2018-01-23 09:43:58.969427: Adjusted maxRetries for 1-20 to 5
2018-01-23 09:43:58.969500: Adjusted maxRetries for 1-28 to 4
2018-01-23 09:43:58.969546: Adjusted maxRetries for 1-34 to 5
2018-01-23 09:43:58.970166: Adjusted maxRetries for 1-132 to 4
…
``` 

I will adjust wiki pages if this gets merged to reflect the changed option names that now include `Mins` as a suffix indicating units.

### CODE REVIEW

File: scripts/task_process/cache_status.py [hunk 1]
```diff
@@ -193,23 +193,40 @@ def parseErrorReport(data, nodes):
...
-            taskStatus['Timestamp'] = ad.get('Timestamp', -1)
-            taskStatus['NodesTotal'] = ad.get('NodesTotal', -1)
-            taskStatus['DagStatus'] = ad.get('DagStatus', -1)
+            if level:
+                statusDict = subDagStatus.setdefault(int(level), {})
+                statusDict['Timestamp'] = ad.get('Timestamp', -1)
+                statusDict['NodesTotal'] = ad.get('NodesTotal', -1)
+                statusDict['DagStatus'] = ad.get('DagStatus', -1)
+            else:
+                dagStatus['Timestamp'] = ad.get('Timestamp', -1)
```
Comment (belforte) [inline scripts/task_process/cache_status.py]: these 3 lines seems to me the same before and after the else. Could they be put out of the if ?

File: src/python/TaskWorker/Actions/Splitter.py [hunk 2]
```diff
@@ -22,28 +22,36 @@ def execute(self, *args, **kwargs):
...
 
+        data = args[0]
         splitparam = kwargs['task']['tm_split_args']
         splitparam['algorithm'] = kwargs['task']['tm_split_algo']
         if kwargs['task']['tm_job_type'] == 'Analysis':
+            totalUnits = kwargs['task']['tm_totalunits']
             if kwargs['task']['tm_split_algo'] == 'FileBased':
-                splitparam['total_files'] = kwargs['task']['tm_totalunits']
+                if totalUnits < 1.0:
+                    totalUnits *= len(data.getFiles())
```
Comment (belforte) [inline src/python/TaskWorker/Actions/Splitter.py]: do we need to round to an integer ?

Comment (belforte) [reply src/python/TaskWorker/Actions/Splitter.py]: also in the following lines

File: scripts/task_process/cache_status.py (hunk 1 above)
Comment (matz-e) [reply scripts/task_process/cache_status.py]: Not sure that I understand your comment? One sets keys in `dagStatus` and one in `statusDict`?

File: src/python/TaskWorker/Actions/Splitter.py (hunk 2 above)
Comment (matz-e) [reply src/python/TaskWorker/Actions/Splitter.py]: I think tentatively not, [as WMCore seems to do this](https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/JobSplitting/FileBased.py#L28).

Comment (belforte) [reply src/python/TaskWorker/Actions/Splitter.py]: well.. unless that int in WMCore is part of their public interace i.e. a clear commitment, I'd rather not rely on it and for mind sanity round to integer here.

File: scripts/task_process/cache_status.py (hunk 1 above)
Comment (belforte) [reply scripts/task_process/cache_status.py]: sorryt. I was confused. You are right. please forgive and forget.

File: src/python/TaskWorker/Actions/Splitter.py (hunk 2 above)
Comment (matz-e) [reply src/python/TaskWorker/Actions/Splitter.py]: Ok

**Comment by matz-e** [comment]
A note on open issues:

* I found at least one corner case where resubmitting did not restart jobs
* Fractional dataset processing needs more work

**Comment by belforte** [comment]
I think we can merge this and close and track those two open issues with additional tickets. Otherwise we never converge, Should not turn an open PR into a branch !

**Comment by matz-e** [comment]
OK. I'll open some two issues.

**Pull request merged**